src/tests/test-ledcontroller.c

```diff
@@ -64,5 +129,482 @@ int main(int argc, char **argv)
+	char *short_opts = NULL ;
+	
+
+	/* Sanity Check input params */
+	if((argv == NULL ) || (*argv == NULL)) 
+	{
+		g_error("No cmdline params found to digest");
+		err = -1 ;
+		goto err_option_digest ;
+	}
+	if( effect == NULL ) 
+	{
+		g_error("cmd-line option digest expects effect structure being passed in"); 
+		err = -1 ;
+		goto err_option_digest ;
+	}
+	if( prog_opts == NULL )
+	{
+		g_error("cmd-line option digest expects effect structure being passed in"); 
+		err = -1 ;
+		goto err_option_digest ;
+	}
+
+	/* init the structure */
+	memset( effect , 0 , sizeof(nyx_led_controller_effect_t));
+
+	/* generate short-option strings */
+	short_opts = generate_option_short_string( prog_opts );
+	if( short_opts == NULL )
+	{
+		g_error("Error generating short-option list for cmd-line params");
+		err = -1 ;
+		goto err_option_digest ;
+	}
+
+	optind = 1 ;
+	for( opt = getopt_long(argc, argv, short_opts, prog_opts , NULL) ;
+	     opt != -1 ;
+	     opt = getopt_long(argc, argv, short_opts, prog_opts , NULL)) 
+	{
+		switch( opt ) 
+		{
+			case '?' : /* intentional fall through */
+			case ':' : /* intentional fall through */
+				g_error("getopt() : Error parsing arguments");
+			case 'h' :
+				usage(*argv);
+				err = -2 ;
+				goto err_option_digest ;
+			case 'd' : 
+				err = map_led_device(optarg , effect);
+				if(err)
+					g_error("Check for valid LED device[%s] failed. err = %d",optarg , err);
+
+				break ;
+			case 'e' : 
+				err = map_led_effect(optarg , effect);
+				if(err)
+					g_error("Check for valid LED device effect[%s] failed. err = %d",optarg , err);
+
+				break ;
+			case 'p' : 
+				err = map_led_effect_param(optarg , effect);
+				if(err)
+					g_error("Check for valid LED params[%s] failed. err = %d",optarg , err);
+
+				break ;
+		}
+	}
+
+	if( argc < 2 )
+	{
+		usage(*argv);
+		err = -2 ;
+		goto err_option_digest ;
+	}
+
+	short_opts = (short_opts != NULL) ? free(short_opts), NULL : NULL ;
+	return err ;
+
+err_option_digest :
+	short_opts = (short_opts != NULL) ? free(short_opts), NULL : NULL ;
+	return err ;
+}
+
+/*! 
+ * \brief         - Check which LED device is being activated
+ * \param[in]     - led - nyx LED device name 
+ * \param[in/out] - nyx_led_controller_effect_t *effect - ptr to led effect requested 
+ * \return        - on success 0 , otherwise -1 
+ */
+static int map_led_device(char *led , nyx_led_controller_effect_t *effect)
+{
+	int err = 0 ;
+
+	/* Sanity check input params */
+	if( led == NULL ) 
+	{
+		g_error("No LED cmd-line param to check : led = NULL");
+		err = -1 ;
+		goto err_led_dev_resolve ;
+	}
+	if( effect == NULL )
+	{
+		g_error("No LED params to map into : effect = NULL");
+		err = -1 ;
+		goto err_led_dev_resolve ;
+	}
+
+	if( strcmp(led , "backlight") == 0 ) 
+	{
+		effect->required.led = NYX_LED_CONTROLLER_BACKLIGHT_LEDS ;
+		effect->backlight.callback = led_controller_cb;
+		effect->backlight.callback_context = NULL;
+	}
+	else if( strcmp(led , "centre") == 0 )
```

Comment:
British(centre) vs American(center)  spellings. Whatever is prefereable :) .. can change if required.